### ISSUE TITLE
LOS 21 failed build: `posix_spawn: Argument list too long`

### ISSUE THREAD

**Issue by georgevdl** [lineageos4microg/docker-lineage-cicd#588]
Used to build fine for LineageOS 20 on the same setup.  
After switching to LineageOS 21 the following error appeared:
```
[ 96% 170707/177033] target Prebuilt: wmshell.protolog.json.gz (out/target/product/pdx203/obj/ETC/wmshell.protolog.json.gz_intermediates/wmshell.protolog.json.gz)
[ 96% 170708/177033] Install: out/target/product/pdx203/system/system_ext/etc/wmshell.protolog.json.gz
posix_spawn: Argument list too long
21:53:34 ninja failed with: exit status 1

#### failed to build some targets (02:11:57 (hh:mm:ss)) ####
```
I run the command
`sudo docker run     -e "BRANCH_NAME=lineage-21.0"     -e "DEVICE_LIST=pdx203"     -e "SIGN_BUILDS=true"     -e "SIGNATURE_SPOOFING=restricted"     -e "WITH_GMS=true"    -e "BUILD_TYPE=user"     -v "/media/Data/LOS/lineage:/srv/src"     -v "/media/Data/LOS/zips:/srv/zips"     -v "/media/Data/LOS/logs:/srv/logs"     -v "/media/Data/LOS/cache:/srv/ccache"     -v "/media/Data/LOS/keys:/srv/keys"     -v "/media/Data/LOS/manifests:/srv/local_manifests"     lineageos4microg/docker-lineage-cicd`
in a VM running Fedora Workstation with 59GB RAM allocated to it.
I tried running `ulimit -s 65536`, `sudo ulimit -s 65536` as well as recompiling the Linux kernel (of the VM) with `#define MAX_ARG_PAGES 64` as suggested by [https://unix.stackexchange.com/questions/45583/argument-list-too-long-how-do-i-deal-with-it-without-changing-my-command](https://unix.stackexchange.com/questions/45583/argument-list-too-long-how-do-i-deal-with-it-without-changing-my-command)  
but the issue persists.  
The exact same issue happens when I build for `fajita`

**Comment by petefoth** [comment]
Thanks. This is the same problem which caused this month's 21.0 builds to fail: see https://github.com/lineageos4microg/docker-lineage-cicd/issues/578#issuecomment-1988259359

**Comment by petefoth** [comment]
I came across the same error when working on #474: there I used `mka dist` as the build command, and that led eventually to the `posix spawn` error. I was not able to fix it, only to 'work around` by using different build targets (see [this commit](https://github.com/lineageos4microg/docker-lineage-cicd/commit/7d3cd5592634e542547042e735b0622f10258426))

My plan is to try a similar workaround fix - replacing the `mka bacon` call in `src/build.sh` with `mka some-other-build-target bacon`. I would welcome ideas about which `some-other-build-target` to try.

**Comment by petefoth** [comment]
I'll try using the `systemimage` target - see [this commit](https://github.com/lineageos4microg/docker-lineage-cicd/commit/bc56c3d2eddaf90d157cb23278e52b3cff79e5f4) in [`pf-fix-posix-spawn-error` branch](https://github.com/lineageos4microg/docker-lineage-cicd/tree/pf-fix-posix-spawn-error)

An image with this 'fix' is on dockerhub if you want to give it a try `docker pull lineageos4microg/docker-lineage-cicd:pf-fix-posix-spawn-error`

**Comment by petefoth** [comment]
That's great - thanks for doing my testing for me 😄 

Please can you tell me what files were in the `zips` directory when the build completed? The files that we need to publish are the main ROM `.zip` file (e.g `[lineage-20.0-20240213-microG-pdx203.zip](https://download.lineage.microg.org/pdx203/lineage-21.0-20240313-microG-pdx203.zip`) and the various `.img` files. We don't really have room to publish the `ota-eng.root.zip` file and our updates use the full ROM `.zip` file.

For us, building the `otapackage` target is just a way of working round the fact that `mka bacon` is broken. I need to know whether `mka otapackage` wil produce all the files we need.

Thanks again

**Comment by petefoth** [comment]
Thanks for that. So `mka otapackage` ***does not*** make the main ROM zip file named `lineage-20.0-20240213-microG-kane.zip` that the `build.sh` expects, and so the script fails and exits :(
It ***does*** make a full ROM zip file & OTA package called `lineage_pdx203-ota-eng.root.zip`, which almost certainly has the same content. So either we need to do some renaming magic in `build.sh` or we find a way of getting `mka` to do that for us.

I'm going to [try using `mka otapackage bacon`](https://github.com/lineageos4microg/docker-lineage-cicd/commit/6451277d6312b862f5e92d527ae57c82eba32c16), which I hope will achieve want we want, but in two steps  rather than the one monolithic step using just `mka bacon` 🤞

**Comment by petefoth** [comment]
> 'm going to [try using `mka otapackage bacon`](https://github.com/lineageos4microg/docker-lineage-cicd/commit/6451277d6312b862f5e92d527ae57c82eba32c16), which I hope will achieve want we want, but in two steps rather than the one monolithic step using just `mka bacon` 🤞

**Dammit! `posix_spawn: Argument list too long
20:38:01 ninja failed with: exit status 1`

Back to dirty renaming spells in `build.sh`

**Comment by petefoth** [comment]
OK - I think this is fixed - see PR  'Fix the posix spawn error #589'

**Comment by petefoth** [comment]
Fixed by PR #589 losing
